packages/web-components/fast-ssr/src/request-storage.ts

```diff
@@ -156,18 +184,30 @@ export const RequestStorageManager = Object.freeze({
 
     /**
      * Installs a DOM shim that ensures that window, document,
-     * and other globals are scoped per-request.
+     * and other globals are scoped per-request. Calling this function
+     * will have no effect if the shim has already been installed.
      */
     installDOMShim(): void {
         for (const key of perRequestGlobals) {
-            const original = (globalThis as any)[key];
-            Reflect.defineProperty(globalThis, key, {
-                get() {
-                    // Return original global variable if currently not in the storage scope
-                    const store = asyncLocalStorage.getStore() as Map<string, any>;
-                    return store ? store.get("window")[key] : original;
-                },
-            });
+            if (!shimIsInstalledFor(key)) {
```

Comment:
In the PR description, it mentioned it could throw when install DOM shim fails. What's the scenario that could happen? Even when `(globalThis as any)[key]` is undefined, the code still won't throw right?

Reply:
installation will fail if, for whatever reason, globals are defined in a non-configurable way prior to installation of the shim.

For example:

```ts
// Define non-configurable window property on global
Object.defineProperty(gloablThis, "window", { value: myWindow, configurable: false });

// Attempts to write descriptor for 'window` on globalThis, which is not configurable
// throws
RequestStorageManger.installDOMShim();

```

Reply:
I'm open to not throwing too, but it seems wrong to fail silently and not make it known that the program could be leaking globals.

Reply:
I am okay with throwing since this makes sure the DOM shim is installed prior to middleware getting initialized.